### ISSUE TITLE
Translations are not working in layouts - Next 13 - App Dir 

### ISSUE THREAD

**Issue by filipenevola** [amannn/next-intl#219]
### Description

If I try to use `getTranslations` or `useTranslations` from a page layout I got the following error message:

`Error: Dynamic server usage: headers`

<img width="981" alt="headers-errors" src="https://user-images.githubusercontent.com/437372/226694617-be5b6099-06cb-4fd9-883f-3f5dfbe99eab.png">

My app is really small (https://news.lemeno.io) so I believe it's not working for all the layouts as my app only has 3 routes.

### Reproduction URL (CodeSandbox)

https://news.lemeno.io

### Reproduction description

Steps to reproduce:

Use getTranslations or useTranslations inside a layout.

### Expected behaviour

No errors.

**Comment by amannn** [comment]
Right, `next-intl` is currently SSR-only as mentioned in [the roadmap](https://next-intl-docs.vercel.app/docs/next-13/server-components#roadmap). The next beta version will include a more helpful error message.

**Comment by amannn** [comment]
The problem you're seeing is related to using `generateStaticParams`—this is mentioned in the roadmap for the Server Components beta and you can look into https://github.com/amannn/next-intl/pull/149 to learn more about the details. 

> Don't you want to keep track of this?

This is already on my list in https://github.com/amannn/next-intl/pull/149 and has been asked a couple of times before.

I understand that this error is frustrating since it's very cryptic, therefore the next beta version will include [a more helpful error message](https://github.com/amannn/next-intl/pull/149/commits/17b245e78f7a1539ba874ecc9c7697a6053190e5) for this case.

I hope this helps!

> BTW, I'm also OSS maintainer in many projects.

That's great to hear! 😊
